pipenv/patched/piptools/repositories/pypi.py

```diff
@@ -65,23 +65,30 @@ def get_hash(self, location):
         new_location = copy.deepcopy(location)
         if orig_scheme in vcs.all_schemes:
             new_location.url = new_location.url.split("+", 1)[-1]
-        can_hash = new_location.hash
-        if can_hash:
-            # hash url WITH fragment
+
+        if new_location.hash is not None:
+            # ignore cache, unless the url contains a hash fragment
             hash_value = self.get(new_location.url)
         if not hash_value:
             hash_value = self._get_file_hash(new_location) if not new_location.url.startswith("ssh") else None
             hash_value = hash_value.encode('utf8') if hash_value else None
-        if can_hash:
-            self.set(new_location.url, hash_value)
+            if new_location.hash is not None:
```

Comment:
What is the difference between this and the original condition?

Reply:
none. The code was hard to read and I attempted to make it more readable.

Reply:
Oh, then `can_hash` should be better since we already defined it before. Maybe change the name to be more meaningful, like `has_hash_fragment`.

Reply:
I eliminated `can_hash` because I found it less readable. The explicit if matches the commentary. 
Would you like me to undo all this?